lang/wasm_alien_thread_runner.hh

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include <functional>
+#include <mutex>
+#include <queue>
+#include <semaphore>
+#include <thread>
+
+#include <seastar/core/future.hh>
+
+#include "rust/cxx.h"
+#include "rust/wasmtime_bindings.hh"
+
+namespace wasm {
+
+struct wasm_compile_task {
+    std::function<void()> func;
+    seastar::promise<rust::Box<wasmtime::Module>>& done;
+    int shard;
+};
+
+struct task_queue {
+    std::mutex _mut = {};
+    bool _stopped = false;
+    static constexpr unsigned QUEUE_SIZE = 128;
+    // Has the same value as _pending.size()
+    std::counting_semaphore<QUEUE_SIZE> _sem{0};
+    std::queue<wasm_compile_task> _pending;
+public:
+    std::optional<wasm_compile_task> pop_front();
+    void stop();
+    void push_back(wasm_compile_task work_item);
+    size_t size() { return _pending.size(); }
+};
+
+class alien_thread_runner {
+    std::thread thread;
+    bool _stopping = false;
+    task_queue pending_queue;
```

Comment:
might want to add `_` prefix to the private member variables to be more consistent with the rest of this project.

Reply:
Yes, I'm surprised I missed that